File: GoldmanSachs/Q5.py
#User function Template for python3
class Solution:
    def getNthUglyNo(self,n):
        # code here
        i2, i3, i5 = 0, 0, 0
        ugly = [0]*(n+1)
        ugly[0] = 1
        next_ugly = 1
        num2, num3, num5 = 2, 3, 5
        for i in range(1, n):
            next_ugly = min(num2, num3, num5)
            ugly[i] = next_ugly
            if next_ugly == num2:
                i2+=1
                num2 = ugly[i2]*2
            if next_ugly == num3:
                i3+=1
                num3 = ugly[i3]*3
            if next_ugly == num5:
                i5+=1
                num5 = ugly[i5]*5
        return next_ugly
        

#{ 
#  Driver Code Starts
#Initial Template for Python 3



if __name__ == '__main__':
    tc = int(input())
    while tc > 0:
        n = int(input())
        ob = Solution()
        ans = ob.getNthUglyNo(n);
        print(ans)
        tc -= 1

# } Driver Code Ends

File: Amazon/Q5.java
// { Driver Code Starts
//Initial Template for Java

import java.io.*;
import java.util.*;

class GFG{
    public static void main(String args[])throws IOException
    {
        BufferedReader in = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(in.readLine());
        while(t-- > 0){
            int n = Integer.parseInt(in.readLine());
            String contact[] = in.readLine().trim().split("\\s+");
            String s = in.readLine();
            
            Solution ob = new Solution();
            ArrayList<ArrayList<String>> ans = ob.displayContacts(n, contact, s);
            for(int i = 0;i < ans.size();i++){
                for(int j = 0;j < ans.get(i).size();j++)
                    System.out.print(ans.get(i).get(j) + " ");
                System.out.println();
            }
        }
    }
}// } Driver Code Ends


//User function Template for Java

class Solution{
    static ArrayList<ArrayList<String>> displayContacts(int n, 
                                        String contact[], String s)
    {
        // code here
        HashSet<String> set = new HashSet<>();
        for (String ele : contact) {
            set.add(ele);
        }
        contact = new String[set.size()];
        int j = 0;
        for (String ele : set) {
            contact[j++] = ele;
        }
        Arrays.sort(contact);
        ArrayList<ArrayList<String>> res = new ArrayList<>();
        for (int i = 0; i < s.length(); i++) {
            res.add(new ArrayList<>());
        }
        
        for (String cont : contact) {
            for (int i = 0; i < cont.length(); i++) {
                if (i < s.length() && s.charAt(i) == cont.charAt(i)) {
                    res.get(i).add(cont);
                } else {
                    break;
                }
            }
        }
        
        for (int i = 0; i < s.length(); i++) {
            if (res.get(i).size() == 0) {
                res.get(i).add("0");
            }
        }
        
        return res;
    }
}